### PR TITLE
boards: arm: nrf5340pdk: adapt shared-memory defines to new style DT

### PR DESCRIPTION
**Pull request by ioannisg**
Adapt shared memory address and size macros to new DT style.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

File: boards/arm/nrf5340pdk_nrf5340/nrf5340_cpunet_reset.c [hunk 1]
```diff
@@ -51,12 +58,13 @@ static int remoteproc_mgr_boot(struct device *dev)
 	remoteproc_mgr_config();
 #endif /* !CONFIG_TRUSTED_EXECUTION_NONSECURE */
 
-#if (DT_IPC_SHM_BASE_ADDRESS != 0)
+#if (SHM_BASE_ADDRESS != 0)
```
Comment (galak) [inline boards/arm/nrf5340pdk_nrf5340/nrf5340_cpunet_reset.c:61]: are you fine w/the assumption that zephyr,ipc-shm is always set?  If not seems like you'll have a case that SHM_BASE_ADDRESS will not be defined if zephyr,ipc-shm doesn't exist.

**Review by galak — APPROVED** [review]

File: boards/arm/nrf5340pdk_nrf5340/nrf5340_cpunet_reset.c (hunk 1 above)
Comment (ioannisg) [reply boards/arm/nrf5340pdk_nrf5340/nrf5340_cpunet_reset.c:61]: zephyr,ipc-shm should always be set, but let's make it more solid and add and #idfef here?

Comment (ioannisg) [reply boards/arm/nrf5340pdk_nrf5340/nrf5340_cpunet_reset.c]: updated it now @galak

**Review by galak — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: boards/arm/nrf5340pdk_nrf5340/nrf5340_cpunet_reset.c [hunk 2]
```diff
@@ -51,12 +58,13 @@ static int remoteproc_mgr_boot(struct device *dev)
...
 
-#if (DT_IPC_SHM_BASE_ADDRESS != 0)
+#if defined(SHM_BASE_ADDRESS) && (SHM_BASE_ADDRESS != 0)
+
 	/* Initialize inter-processor shared memory block to zero. It is
 	 * assumed that the application image has access to the shared
 	 * memory at this point (see #24147).
 	 */
-	memset((void *) DT_IPC_SHM_BASE_ADDRESS, 0, KB(DT_IPC_SHM_SIZE));
+	memset((void *) SHM_BASE_ADDRESS, 0, SHM_SIZE);
```
Comment (mbolivar-nordic) [inline boards/arm/nrf5340pdk_nrf5340/nrf5340_cpunet_reset.c:67]: Don't you need KB(SHM_SIZE)?

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Just putting a -1 on this so it doesn't get merged until my question is resolved.

If KB() is not needed, the commit log should say why

**Comment by ioannisg** [comment]
@mbolivar the old-style DT macro `DT_IPC_SHM_SIZE ` is expressing the size in KB. (The value is 0x40.) That's why we need the KB macro in `memset`.

The new-style DT macro is expressing the actual size. (The value is 0x10000). That's why we don't need the KB macro.

Since this is all normal and bug-free, I don't think the commit message should explain any of this. Switching to using the new macros is all we are doing here.

**Review by mbolivar-nordic — APPROVED** [review]
Sorry, missed the ping to @mbolivar

**Comment by ioannisg** [comment]
> Sorry, missed the ping to @mbolivar

My bad @mbolivar-nordic - we should adapt to your new handle :)

**Pull request merged**